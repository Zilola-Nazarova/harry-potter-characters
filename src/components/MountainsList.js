import { useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import { BsArrowRightCircle } from 'react-icons/bs';
import styles from '../styles/MountainsList.module.css';
import weatherCodes from '../database/weatherCodes';

const MountainsList = () => {
  const { mountains, isLoading, error } = useSelector((store) => store.mountains);
  const [inputText, setInputText] = useState('');

  const inputHandler = (e) => {
    // convert input text to lower case
    const lowerCase = e.target.value.toLowerCase();
    setInputText(lowerCase);
  };

  const filteredMountains = mountains.filter((peak) => {
    // if no input the return the original
    if (inputText === '') {
      return peak;
    }
    // return the item which contains the user input
    return peak.name.toLowerCase().includes(inputText);
  });

  return (
    <>
      <div className={styles.search}>
        <h3>mountain peaks</h3>
        <input
          placeholder="search..."
          onChange={inputHandler}
        />
      </div>
      {(isLoading) && <p className="loading">The mountains list is loading...</p>}

      {(error) && (
        <p className="error">
          Something went wrong!
          <br />
          { error }
        </p>
      )}

      {(!isLoading && !error) && (
        <>
          {(filteredMountains.length === 0) && (
            <p className={styles.error}>
              Something went wrong!
              <br />
              No matching mountains found.
            </p>
          )}

          <ul className={styles.list}>
            {filteredMountains.sort((a, b) => a.name.localeCompare(b.name)).map((peak) => (
              <li
                key={uuidv4()}
                className={styles.card}
              >
                <Link to={`details/${peak.name}`}>
                  <BsArrowRightCircle className={styles.icon} />
                  <img alt="mountain peak" src={peak.image} />
                  <div className={styles.info}>
                    <img
                      className={styles.code}
                      alt={peak.current_weather.weathercode}
                      src={weatherCodes[peak.current_weather.weathercode]}
                    />
                    <div>
                      <h4 className={styles.name}>{peak.name}</h4>
                      <span className={styles.elevation}>{peak.elevation}</span>
                    </div>
                  </div>
                </Link>
              </li>
            ))}
          </ul>
        </>
      )}
    </>
  );
};

export default MountainsList;
