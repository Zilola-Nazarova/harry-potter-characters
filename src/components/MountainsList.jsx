import styles from '@/styles/MountainsList.module.css';
import { Link } from "react-router-dom";

const MountainsList = ({data}) => {
  const { mountains, isLoading, error } = data;
  if (isLoading) {
    return <p className={styles.loading}>The forecast is loading!</p>;
  }
  if (error) {
    return (
      <p className={styles.error}>
        Something went wrong!
        <br />
        { error }
      </p>
    );
  }
  if (mountains.length !== 0) {
    return (
      <div>
        {
          data.mountains.map((peak) => (
            <li key={peak.name}>
              <p>{peak.name}</p>
              <p>{peak.elevation}</p>
              <p>{JSON.stringify(peak.current_weather)}</p>
              <Link to={`details/${peak.name}`}>See Details -&gt;</Link>
            </li>
          ))
        }
      </div>
    )
  }
}

export default MountainsList;
