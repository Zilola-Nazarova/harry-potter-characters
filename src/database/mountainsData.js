const mountainsData = [
  {
    name: 'Greater Chimgan',
    latitude: 41.4951900, 
    longitude: 70.0586718,
    elevation: 3309.6,
    current_weather: {
      temperature: null,
      windspeed: null,
      windderection: null,
      time: null,
    }
  },
  {
    name: 'Karakush',
    latitude: 41.2477890, 
    longitude: 70.3391467,
    elevation: 3864.3,
    current_weather: {
      temperature: null,
      windspeed: null,
      windderection: null,
      time: null,
    }
  },
  {
    name: 'Babaitag',
    latitude: 41.1549650,
    longitude: 70.2232660,
    elevation: 3555.7,
    current_weather: {
      temperature: null,
      windspeed: null,
      windderection: null,
      time: null,
    }
  },
  {
    name: 'Some place',
    latitude: 41.1549650,
    longitude: 75.2232660,
    elevation: 3555.7,
    current_weather: {
      temperature: null,
      windspeed: null,
      windderection: null,
      time: null,
    }
  },
]

export default mountainsData;
