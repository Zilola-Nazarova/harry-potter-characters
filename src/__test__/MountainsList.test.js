import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import configureStore from 'redux-mock-store';
import MountainsList from '../components/MountainsList';
import mountainsData from '../database/mountainsData';

test('Check if Missions component is rendered correctly', () => {
  const applyMiddleware = [thunk, logger];
  const mockingStore = configureStore(applyMiddleware);
  const store = mockingStore({
    mountains: mountainsData,
    isLoading: false,
    error: false,
  });

  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <MountainsList />
      </BrowserRouter>
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
});
