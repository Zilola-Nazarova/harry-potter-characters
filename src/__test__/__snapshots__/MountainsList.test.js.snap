// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Missions component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="search"
      >
        <h3>
          mountain peaks
        </h3>
        <input
          placeholder="search..."
        />
      </div>
      <p
        class="error"
      >
        Something went wrong!
        <br />
        No matching mountains found.
      </p>
      <ul
        class="list"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="search"
    >
      <h3>
        mountain peaks
      </h3>
      <input
        placeholder="search..."
      />
    </div>
    <p
      class="error"
    >
      Something went wrong!
      <br />
      No matching mountains found.
    </p>
    <ul
      class="list"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
